tools/oilOutpost.js

```diff
@@ -311,18 +311,21 @@ module.exports = function(string, opt = {}) {
   let trainStopLocation = null;
 
   if (INCLUDE_TRAIN_STATION) {
-    trainStopLocation = generateTrainStation(bp, { x: target.x + 3 + 3 * TANKS, y: target.y - 2 }, Math.max(bp.bottomRight().y, target.y - 2 -
-      WAGONS * 7), {
+    trainStopLocation = generateTrainStation(bp,
+       { x: target.x + 3 + 3 * TANKS, y: target.y - 2 },
+        Math.max(bp.bottomRight().y, target.y + 2 + WAGONS * 7 + ((SINGLE_HEADED_TRAIN ? 0 : 1) * LOCOMOTIVES)), 
```

Comment:
This seemed to be the issue with train rails being too short

Reply:
Trains have been the biggest headache of all the issues in this project :P